lib/puppet/provider/keystore_certificate/openssl.rb

```diff
@@ -0,0 +1,95 @@
+Puppet::Type.type(:keystore_certificate).provide(:openssl) do
+  commands :openssl => 'openssl'
+  commands :keytool => 'keytool'
+
+  def create
+    add_certificate
+  end
+
+  def destroy
+    delete_certificate
+  end
+
+  def exists?
+    keystore_content
```

Comment:
I don't think this is a valid thing. I'd expect that `keystore_content` always returns a string which evaluates to true. Or does it rely on it raising a `Puppet::ExecutionFailure`if the alias isn't present?

Reply:
The latter. We have two existence conditions currently:

1) Does the specified alias exist in the specified keystore
2) Does the specified certificate fingerprint match what is in the keystore